docs/dev/core/internal/payload.md

```diff
@@ -11,7 +11,111 @@ It is written as a set of [templates](https://github.com/mozilla-services/mozill
 
 ## Metric types
 
-TODO: Fill in the rest of the metric types. https://bugzilla.mozilla.org/show_bug.cgi?id=1566854
+### Boolean
+
+A [Boolean](../../../user/metrics/boolean.md) is represented by its boolean value.
+
+#### Example
+
+```json
+true
+```
+
+
+### Counter
+
+A [Counter](../../../user/metrics/counter.md) is represented by its integer value.
+
+#### Example
+
+```json
+17
+```
+
+### String
+
+A [String](../../../user/metrics/string.md) is represented by its string value.
+
+#### Example
+
+```json
+"sample string"
+```
+
+### String list
+
+A [String List](../../../user/metrics/string_list.md) is represented as an array of strings.
+
+```json
+["sample string", "another one"]
+```
+
+### Timespan
+
+A [Timespan](../../../user/metrics/timespan.md) is represented as an object of their duration as an integer and the time unit.
+
+| Field name | Type | Description |
+|---|---|---|
+| `value` | Integer | The value in the marked time unit. |
+| `time_unit` | String | The time unit, see the [timespan's configuration](../../../user/metrics/timespan.md#configuration) for valid values. |
+
+#### Example
+
+```json
+{
+    "time_unit": "milliseconds",
+    "value": 10
+}
+```
+
+## Timing Distribution
+
+A [Timing distribution](../../../user/metrics/timing_distribution.md) is represented as an object with the following fields.
+
+| Field name | Type | Description |
+|---|---|---|
+| `bucket_count` | Integer | The bucket count of the histogram. |
+| `range` | Array&lt;Integer&gt; | The range indicated by its minimum and maxium value. |
+| `sum` | Integer | The sum of all recorded values. |
+| `time_unit` | String | The timespan's time unit, see the [time distribution's configuration](../../../user/metrics/timing_distribution.md#configuration) for valid values. |
+| `values` | Map&lt;String, Integer&gt; | The values in each bucket. The key is the bucket's minimum value. Empty buckets are removed. |
```

Comment:
Mh, this still gets me a bit confused after having removed the context. What do you think about something like this: "The values in each bucket. The key is the minimum value for the range of that bucket. Buckets with no values are not reported"